Hot-Rod-Transactions-NonXA.asciidoc

```diff
@@ -0,0 +1,313 @@
+
+At the first operation, the remote cache registers itself as a participant in the `SynchronizationAdapter`.
+
+=== Transaction execution
+
+Read operations is executed as following:
+
+* If the key exists in transaction context, return its value;
+* Otherwise, fetch the value and metadata from the server and store it in the transaction context.
+
+Write operations is executed as following:
+
+* If the key doesn't exists in the transaction context and it is required (conditional value or needs to return previous value), fetches value and metadata from the server and store it in the transaction context;
+* Applies the operation based on the value in the transaction context and updates its value.
+
+NOTE: The value and metadata can be fetch from any node in the cluster.
+
+=== Transaction commit
+
+A `Transaction` is committed in 2 phases: `beforeCompletion()` and `afterCompletion()`.
+
+In `beforeCompletion()`, the client collects the written keys and sends them to the server (any member in cluster).
+The following information is sent:
+
+* XID - this is the transaction id;
+* For each key, it sends:
+** The key;
+** The `ReadState`;
+** The version read (if any);
+** The new value;
+** The metadata;
+
+[NOTE]
+======
+The `ReadState` represents the following information:
+
+* `NOT_READ`: the key wasn't read before the write;
+* `NON_EXISTING`: the key was read before write and it didn't exist;
+* `READ`: The key was read before write and it exists (the versionr read is required)
+======
+
+In the server side, when the request is received, it replays the transaction.
+A new transaction is created and set its XID based on the XID received (to avoid any XID translation).
+
+For each key received, the server performs the following:
+
+* It checks the `ReadState` value:
+** `NOT_READ`: it doesn't fetch the previous value.
+** `NON_EXISTING`: it checks if the key doesn't exist.
+If the key exists, the transaction is rolled-back.
+** `READ`: it fetches the value and metadata.
+Then it compares the version received with the version stored in the metadata.
+If they don't match, the transaction is rolled-back.
+* Finally, it performs the write.
+
+When all keys are handled, the server invokes `EmbeddedTransaction.runPrepare()`.
+The outcome is sending back to the client.
+
+The client receives the outcome and marks the transaction to rollback if the server failed to prepare.
+
+Finally, in `afterCompletion()`, the client only sends the commit or rollback request:
+
+* XID
+* `COMMIT` or `ROLLBACK` request.
+
+In the server, if the transaction was executed locally, it invokes the `EmbeddedTransaction.runCommit(`).
+Otherwise, use the `RecoveryManager` to commit or rollback the transaction.
+
+NOTE: Not sure if the `RecoveryManager` can be used or if I need to create a Global Transaction Table.
+
+== Hot Rod Protocol Details
+
+=== Operation Codes
+
+4 new operation codes:
+
+|===
+|Operation Code | Response Operation Code |Operation
```

Comment:
I agree, it should be considered whether the commands for tx caches could be implemented using server tasks.